cmd/cli/main.go

```diff
@@ -14,6 +14,151 @@
+}
 
+const port = "8080"
+
+var tagCmd = &cobra.Command{
+	Use:   "tag create tag_name commit_sha [-a author] [-e email]\n   or: tag delete tag_name",
+	Short: "tags commits in the repository",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return fmt.Errorf("too few args")
+		}
+		if args[0] == "create" {
+			if len(args) != 3 {
+				return fmt.Errorf("unexpected number of args for tag create")
+			}
+		} else if args[0] == "delete" {
+			if len(args) != 2 {
+				return fmt.Errorf("unexpected number of args for tag delete")
+			}
+		} else {
+			return fmt.Errorf("unsupported keyword (not create or delete)")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		url := "http://localhost:" + port + "/tag"
+		var request types.TagRequest
+		if args[0] == "create" {
+			request = types.TagRequest{
+				Type:   types.TagCreate,
+				Tag:    args[1],
+				Sha:    args[2],
+				Author: tagAuthor,
+				Email:  tagEmail,
+			}
+		} else {
+			request = types.TagRequest{
+				Type: types.TagDelete,
+				Tag:  args[1],
+			}
+		}
+		j, err := json.Marshal(request)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		resp, err := http.Post(url, "application/json", bytes.NewBuffer(j))
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		if resp.StatusCode == http.StatusBadRequest || resp.StatusCode == http.StatusInternalServerError {
+			fmt.Println("Error encountered while processing tag request")
+			return
+		}
+		fmt.Println(string(body))
+	},
+}
+
+var rollbackCmd = &cobra.Command{
+	Use:   "rollback -t tag_name | -s commit_sha",
+	Short: "rollback to the specified commit by tag name or SHA",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return fmt.Errorf("unexpected number of args for rollback")
+		}
+		if (rollbackTag != "") == (rollbackSHA != "") {
+			return fmt.Errorf("must specify exactly one of -t or -s")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		url := "http://localhost:" + port + "/rollback"
+		request := types.RollbackRequest{
+			Tag: rollbackTag,
+			Sha: rollbackSHA,
+		}
+		j, err := json.Marshal(request)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		resp, err := http.Post(url, "application/json", bytes.NewBuffer(j))
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		if resp.StatusCode == http.StatusBadRequest || resp.StatusCode == http.StatusInternalServerError {
+			fmt.Println("Error encountered while processing rollback request")
+			return
+		}
+		fmt.Println(string(body))
+	},
+}
+
+func init() {
+	tagCmd.Flags().StringVarP(&tagAuthor, "author", "a", "no-author", "tag author")
```

Comment:
I think this should work based off a kubeconfig (either in-cluster config, or through the `KUBECONFIG` env variable / `--kubeconfig` flag). It should replace both the author and email flag IMO and it will be more consistent with how we handle / commit audit events.

Reply:
more investigation needed